tests/ginkgo/sequential/1-051_validate_argocd_agent_principal_test.go

```diff
@@ -351,28 +356,31 @@ var _ = Describe("GitOps Operator Sequential E2E Tests", func() {
 						Namespace: ns.Name,
 					},
 				}
-				Eventually(service).Should(k8sFixture.ExistByName())
-				Expect(string(service.Spec.Type)).To(Equal("ClusterIP"))
+				Eventually(service).Should(k8sFixture.ExistByName(), "Service '%s' should exist in namespace '%s'", serviceName, ns.Name)
+
+				// skip principal service
+				if serviceName != argoCDAgentPrincipalName {
+					Expect(string(service.Spec.Type)).To(Equal("ClusterIP"), "Service '%s' should have ClusterIP type, got '%s'", serviceName, service.Spec.Type)
+				}
 			}
 
-			By("verifying Route for principal exists on OpenShift")
+			if shouldExpectRoute {
+				By("verifying Route for principal exists on OpenShift")
```

Comment:
Nit: IMHO can move this `By` inside the `if fixture.RunningOnOpenShift {}`,so as not to confuse the case where it is NOT running on openshift